evennia/contrib/base_systems/components/tests.py

```diff
@@ -50,73 +68,80 @@ class TestComponents(EvenniaTest):
     character_typeclass = CharacterWithComponents
 
     def test_character_has_class_components(self):
-        assert self.char1.test_a
-        assert self.char1.test_b
+        self.assertTrue(self.char1.test_a)
+        self.assertTrue(self.char1.test_b)
 
     def test_inherited_typeclass_does_not_include_child_class_components(self):
         char_with_c = create.create_object(
             InheritedTCWithComponents, key="char_with_c", location=self.room1, home=self.room1
         )
-        assert self.char1.test_a
-        assert not self.char1.cmp.get("test_c")
-        assert char_with_c.test_c
+        self.assertTrue(self.char1.test_a)
+        self.assertFalse(self.char1.cmp.get("test_c"))
+        self.assertTrue(char_with_c.test_c)
 
     def test_character_instances_components_properly(self):
-        assert isinstance(self.char1.test_a, ComponentTestA)
-        assert isinstance(self.char1.test_b, ComponentTestB)
+        self.assertIsInstance(self.char1.test_a, ComponentTestA)
+        self.assertIsInstance(self.char1.test_b, ComponentTestB)
 
     def test_character_assigns_default_value(self):
-        assert self.char1.test_a.my_int == 1
-        assert self.char1.test_a.my_list == []
+        self.assertEquals(self.char1.test_a.my_int, 1)
+        self.assertEquals(self.char1.test_a.my_list, [])
 
     def test_character_assigns_default_provided_values(self):
-        assert self.char1.test_b.my_int == 3
-        assert self.char1.test_b.my_list == [1, 2, 3]
+        self.assertEquals(self.char1.test_b.my_int, 3)
+        self.assertEquals(self.char1.test_b.my_list, [1, 2, 3])
```

Comment:
`my_list` is configured with autocreate=True, but to make sure this is actually created in the db, one need to make sure this is actually available from the AttributeHandler as well.

Another classic test here is to make sure to check what happens if you do `self.char1.test_b.my_list.append(4)` and make sure behavior is as expected.

Reply:
I am adding `test_character_has_autocreated_values ` in my next commit.
For the append test, I think `test_mutables_are_not_shared_when_autocreate` already tests that.